src/components/file-preview.js

```diff
@@ -0,0 +1,214 @@
+    <div className="FilePreview-icon"><FileOutlined /></div>
+  );
+
+  let renderPreview;
+  switch (resourceType) {
+    case RESOURCE_TYPE.audio:
+      renderPreview = renderAudio;
+      break;
+    case RESOURCE_TYPE.video:
+      renderPreview = renderVideo;
+      break;
+    case RESOURCE_TYPE.image:
+      renderPreview = renderImage;
+      break;
+    case RESOURCE_TYPE.pdf:
+      renderPreview = renderPdf;
+      break;
+    case RESOURCE_TYPE.text:
+      renderPreview = renderText;
+      break;
+    default:
+      renderPreview = renderGenericFile;
+  }
+
+  return (
+    <div className="FilePreview">
+      <div className="FilePreview-headerArea">
+        {fileName}
+      </div>
+      <div className="FilePreview-previewArea">
+        {renderPreview()}
+      </div>
+      <div className="FilePreview-detailsArea">
+        <div className="FilePreview-detailLabel">
+          {t('common:name')}
+        </div>
+        <div className="FilePreview-detailValue">
+          {fileName}
+        </div>
+        <div className="FilePreview-detailLabel">
+          {t('common:type')}
+        </div>
+        <div className="FilePreview-detailValue">
+          {mimeTypeHelper.localizeCategory(category, t)}
+        </div>
+        <div className="FilePreview-detailLabel">
+          {t('common:size')}
+        </div>
+        <div className="FilePreview-detailValue">
+          {prettyBytes(size)}
+        </div>
+        <div className="FilePreview-detailLabel">
+          {t('common:createdOn')}
+        </div>
+        <div className="FilePreview-detailValue">
+          {formatDate(lastModified)}
+        </div>
+        {imageDimensions && (
+          <Fragment>
+            <div className="FilePreview-detailLabel">
+              {t('common:naturalSize')}
+            </div>
+            <div className="FilePreview-detailValue">
+              {imageDimensions.width}&nbsp;⨉&nbsp;{imageDimensions.height}&nbsp;px
+            </div>
+          </Fragment>
+        )}
+        {pdf && (
+          <Fragment>
+            <div className="FilePreview-detailLabel">
+              {t('common:pageCount')}
+            </div>
+            <div className="FilePreview-detailValue">
+              {pdf.numPages}
+            </div>
+          </Fragment>
+        )}
+        <div className="FilePreview-detailLabel">
+          {t('common:url')}
+        </div>
+        <div className="FilePreview-detailValue">
+          <LiteralUrlLink rel="noopener noreferrer" target="_blank" href={url} />
+          &nbsp;&nbsp;
+          <Tooltip title={t('common:copyUrlToClipboard')}>
+            <a onClick={handleCopyUrlToClipboardClick}><CopyToClipboardIcon /></a>
+          </Tooltip>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+FilePreview.propTypes = {
+  lastModified: PropTypes.instanceOf(Date).isRequired,
+  size: PropTypes.number.isRequired,
+  url: PropTypes.string.isRequired
+};
+
+export default FilePreview;
```

Comment:
Super nice preview setup, seeing all the info there makes me realize this will be an extremely useful piece of transparency added to the CDN resources management 👍 

From UI perspective, I would have a suggestion for you to check out and see how you feel about it:
<img width="1224" alt="Screen Shot 2022-05-17 at 10 19 24" src="https://user-images.githubusercontent.com/8189923/168765071-df32dcf5-bf4b-41b1-b4c4-704c95ff6624.png">

Reply:
I like it better next to the label, thanks, good suggestion, I moved it.

Reply:
Nice 👍 
What do you think about the other changes in the screenshot?
- more space between the preview and the text column
- removing the file name from the top
- inverting the background colors (white for preview and grey for text)
- centering the text column vertically

Reply:
Ha! Now I see the other changes :rofl:

Yeah, that looks great, I'll do them. I don't think we should set the gray background color for the preview component though, that might look weird when placed somewhere as it is not a confined block. We can always do that in the parent container (I'll make the preview box have an explicit white background though in order to stand out).

Reply:
Done, now it looks like this:

![file-preview](https://user-images.githubusercontent.com/2676617/168808520-ce79a86b-9524-4416-b169-d5e55932fe55.png)